web/src/components/navbar/NavBar.test.tsx

```diff
@@ -172,36 +195,43 @@ describe("<NavBar />", () => {
       expect(screen.queryByText(Config.navigationDefaults.logInButton)).not.toBeInTheDocument();
     });
 
+    it("displays myNJ button when user is authenticated", async () => {
+      useMockUserData({});
+      renderDesktopNav();
+      expect(screen.queryByText(Config.navigationDefaults.profileLinkText)).not.toBeInTheDocument();
+    });
+
+    it("displays log out button when user is authenticated", async () => {
+      useMockUserData({});
+      renderDesktopNav();
+      expect(screen.queryByText(Config.navigationDefaults.logInButton)).not.toBeInTheDocument();
+    });
+
     it("displays a closed dropdown menu on the NavBar", () => {
-      const user = { name: "John Smith", email: "test@example.com" };
+      const userData = generateBusinessNamedUserData();
+      useMockUserData({ ...userData });
 
-      useMockUserData({
-        user: generateUser(user),
-      });
       renderDesktopNav();
-      const menuEl = screen.getByText(user.name);
+      const menuEl = screen.getByText(getNavBarBusinessTitle(userData));
       expect(menuEl).toBeInTheDocument();
       expect(screen.queryByText(Config.navigationDefaults.logoutButton)).not.toBeInTheDocument();
       expect(screen.queryByText(Config.navigationDefaults.myNJAccountText)).not.toBeInTheDocument();
     });
 
     it("displays an open dropdown menu when clicked and closes when clicked again", async () => {
-      const user = { name: "John Smith", email: "test@example.com" };
-
-      useMockUserData({
-        user: generateUser(user),
-      });
+      const userData = generateBusinessNamedUserData();
+      useMockUserData({ ...userData });
```

Comment:
I see you changed a one-liner to this two-liner in many of these tests. Is there a reason to add the temp variable? Why not just `useMockUserData(generateBusinessNamedUserData())`

Reply:
So the reason is that we need access to the userData to determine the name of the menu depending on businessName or industryId/LegalStructure and sometimes the names of buttons as well based on that